wlog/wapp/fatal.go

```diff
@@ -32,13 +32,23 @@ func RunWithFatalLogging(ctx context.Context, runFn func(ctx context.Context) er
 			return
 		}
 		stacktrace := diag1log.ThreadDumpV1FromGoroutines(debug.Stack())
-		svc1log.FromContext(ctx).Error("panic recovered",
-			svc1log.SafeParam("stacktrace", stacktrace),
-			svc1log.UnsafeParam("recovered", r))
-		if retErr == nil {
-			retErr = werror.Error("panic recovered",
-				werror.SafeParam("stacktrace", stacktrace),
-				werror.UnsafeParam("recovered", r))
+		if err, ok := r.(error); ok {
+			svc1log.FromContext(ctx).Error("panic recovered",
+				svc1log.SafeParam("stacktrace", stacktrace),
+				svc1log.Stacktrace(err))
+			if retErr == nil {
+				retErr = werror.Wrap(err, "panic recovered",
+					werror.SafeParam("stacktrace", stacktrace))
+			}
+		} else {
+			svc1log.FromContext(ctx).Error("panic recovered",
+				svc1log.SafeParam("stacktrace", stacktrace),
+				svc1log.UnsafeParam("recovered", r))
+			if retErr == nil {
+				retErr = werror.Error("panic recovered",
+					werror.SafeParam("stacktrace", stacktrace),
+					werror.UnsafeParam("recovered", r))
+			}
```

Comment:
couldn't come up with a prettier way to do this; feels like a good deal of repeated strings and such. Feedback welcome 🤷‍♂️